### PR TITLE
Add NegationToggle component

### PR DESCRIPTION
**Pull request by micgro42**
The tests will be added when we start using the ButtonToggle component from the Design System.

Bug: [T272011](https://phabricator.wikimedia.org/T271788)

### CODE REVIEW

File: src/components/NegationToggle.vue [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<template>
+	<span class="negationToggle">
+		<Button
+			:type="value==='with' ? 'progressive' : 'neutral'"
```
Comment (guergana) [inline src/components/NegationToggle.vue:4]: it doesn't seem to change to progressive (as in blue background). when it is selected it only shows a blue border around the selected button. is this the desired effect?

File: src/components/NegationToggle.vue [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			:variant="value==='with' ? 'primary' : 'normal'"
+			@click.native="$emit('input', 'with')"
+		>{{ $i18n( 'query-builder-negation-option-label-with' ) }}</Button>
+		<Button
+			:type="value==='without' ? 'progressive' : 'neutral'"
+			:variant="value==='without' ? 'primary' : 'normal'"
```
Comment (guergana) [inline src/components/NegationToggle.vue:10]: same as previous comment. doesn't change the class. the button is always normal

**Review by guergana — DISMISSED** [review]
Doesnt change the class when one of the buttons is clicked.

**Review by guergana — COMMENTED** [review]
Looks good. Tho we might be relegating a lot into the QueryCondition Component.

**Comment by guergana** [comment]
Has to be rebased and I think it's good to go. :)

**Review by guergana — APPROVED** [review]

**Comment by micgro42** [comment]
Rebased on top of master :+1:

**Pull request merged**